content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+
+In the constructor, you declared a constructor passing in @props@, called the super constructor of the @props@. You declared three items in your state assignment namely @msg@ , @txt@ and @user@.
+
+Next, you defined the @componentDidMount@ method which @React@ will fire once the component is mounted. In this method, you used the @AsyncStorage@ component to check if a value exists and if it does, set the state of @user@ to that value. Also, you called the @subscribe@  function which we will treat soon.
+
+The @subscribe@ method is where all real-time subscriptions such as @presence@ and @subscribe@ are made. Let’s look at what goes on in this method.
+
+# First, you set the @ably@ variable to an instance of the real-time library passing in your API Key and the current user. The reason for putting in the current user is because you want to access presence events.
+# You made a subscription to a channel called @cog-tie@.
+# You made a subscription to the presence event of @enter@ which notifies that a new member has joined the chat (by calling @channel.presence.subscribe@). You then created a message object from the response and append to the @msg@ state.
+# You made a subscription to the presence event of @leave@ which notifies that a member has left the chat.
+# You made a call to @channel.presence.enter()@ which triggers the entry into the presence event of the channel been listened to.
+# Next, a normal subscription is made to the @message@ event which emits real-time chat messages, and the message is added to the array of messages available and displayed in real-time.
+
+In the @load_message@ function, a lot of things are going on here, but what the function primarily does is to loop through the messages, check for the action of the messages, and apply either the chat style or notification style to the messages. For example, messages from the presence event @leave@ and @enter@ have the same styling, while messages from the @message@ event have a chat bubble style. Also, the loop checks for the user who sent the message and floats the message to the appropriate position. For example, messages sent from you will be floated to the left, while other users messages would be floated to the right. 
+
+In the @submitChat@ function, you trigger a @POST@ request to our API which in turns publishes the message to the right channel and event. You used the @fetch@ API provided by React Native to send an AJAX request to our server which we created earlier.
+
+p(tip). Note: If you use IP address such as @127.0.0.1@ or @localhost@, the request is most likely going to fail. This is because, in React Native, @localhost@ or @127.0.0.1@ refers to the internal application. Please use the network IP for your system instead.
```

Comment:
Can we provide a link to an article / example of how to do this online pelase?

Reply:
Fixed, thanks